lib/pbench/agent/results.py

```diff
@@ -123,60 +206,143 @@ def make_result_tb(self) -> str:
+                        [self.xz_path, "-T0"],
+                        cwd=str(self.target_dir),
+                        stdin=subprocess.PIPE,
+                        stdout=ofp,
+                        stderr=efp,
+                    )
+                    tar_proc = subprocess.Popen(
+                        args,
+                        cwd=str(self.result_dir.parent),
+                        stdin=None,
+                        stdout=xz_proc.stdin,
+                        stderr=efp,
+                    )
+                    tar_proc.wait()
+                    # Now that the `tar` command has exited, we close the
+                    # `stdin` of the `xz` command to shut it down.
+                    xz_proc.stdin.close()
+                    xz_proc.wait()
+        except Exception as exc:
+            msg = self._unlink_tarball(
+                tarball,
+                f"Tar ball creation failed for {self.result_dir}, skipping {exc}",
             )
-            try:
-                tarball.unlink()
-            except Exception as exc:
-                if not isinstance(exc, OSError) or exc.errno != errno.ENOENT:
-                    raise RuntimeError(
-                        f"Error removing failed tarball, '{tarball}', {exc}"
+            raise RuntimeError(msg)
+        else:
+            if tar_proc.returncode == 0:
+                if xz_proc is not None and xz_proc.returncode != 0:
+                    msg = self._unlink_tarball(
+                        tarball,
+                        f"Failed to create tar ball; 'xz' return code: {xz_proc.returncode:d}",
                     )
+                    raise RuntimeError(msg)
+                else:
+                    try:
+                        e_file.unlink()
+                    except FileNotFoundError:
+                        pass
+                    except Exception as unlink_exc:
+                        self.logger.warning(
+                            "Failed to remove 'tar' stderr file, %s: '%s'",
+                            e_file,
+                            unlink_exc,
+                        )
+            else:
+                # We explicitly ignore the return code from the optional 'xz' process.
+                msg = self._unlink_tarball(
+                    tarball,
+                    f"Failed to create tar ball; 'tar' return code: {tar_proc.returncode:d}",
+                )
+                raise RuntimeError(msg)
+        try:
+            (tar_len, tar_md5) = md5sum(tarball)
+        except Exception:
+            msg = self._unlink_tarball(
+                tarball,
+                f"Failed to verify and generate MD5 for created tar ball, '{tarball}'",
+            )
+            raise RuntimeError(msg)
 
         # The contract with the caller is to just return the full path to the
-        # created tarball.
-        return str(tarball)
+        # created tar ball.
+        return TarballRecord(tarball, tar_len, tar_md5)
```

Comment:
Since you went to the trouble of creating a named tuple, you should use the names:
```
        return TarballRecord(name=tarball, length=tar_len, md5=tar_md5)
```